react/ProductListContext.tsx

```diff
@@ -53,8 +81,17 @@ const initialState: State = {
   sentIds: new Set(),
 }
 
-const ProductListProvider: FC = ({ children }) => {
+interface ProviderProps {
+  listName: string
+}
+
+const ProductListProvider: FC<ProviderProps> = ({ children, listName }) => {
   const [state, dispatch] = useReducer(productListReducer, initialState)
+
+  useEffect(() => {
+    dispatch({ type: 'SET_LIST_NAME', args: { listName } })
+  }, [listName])
```

Comment:
I think we should add a comment here saying that because of this we can't use the same context for two different components (eg: shelf and search result).  I feel that this can happen in the future and adding the comment will help us discover what happened.